rest/client.go

```diff
@@ -34,10 +34,13 @@ const (
 	httpPut    = "PUT"
 )
 
+// The type that is the function that is to be wrapped by the ClientMiddleware
 type InvocationHandler func(c *http.Client, method, url string, body interface{}, header http.Header) (*Response, error)
```

Comment:
Thanks for the link, I've updated the docstrings and will try and keep it in mind for future use.